gui-agent/vmside.c

```diff
@@ -1505,54 +1504,77 @@ static void mkghandles(Ghandles * g)
-    g->net_wm_name = XInternAtom(g->display, "_NET_WM_NAME", False);
-    net_supporting_wm_check = XInternAtom(g->display, "_NET_SUPPORTING_WM_CHECK", False);
-    net_supported = XInternAtom(g->display, "_NET_SUPPORTED", False);
-    supported[0] = net_supported;
-    supported[1] = net_supporting_wm_check;
-    /* _NET_WM_MOVERESIZE required to disable broken GTK+ move/resize fallback */
-    supported[2] = XInternAtom(g->display, "_NET_WM_MOVERESIZE", False);
-    supported[3] = XInternAtom(g->display, "_NET_WM_STATE", False);
-    supported[4] = XInternAtom(g->display, "_NET_WM_STATE_FULLSCREEN", False);
-    supported[5] = XInternAtom(g->display, "_NET_WM_STATE_DEMANDS_ATTENTION", False);
-    XChangeProperty(g->display, g->stub_win, g->net_wm_name, g->utf8_string_atom,
-            8, PropModeReplace, (unsigned char*)"Qubes", 5);
-    XChangeProperty(g->display, g->stub_win, net_supporting_wm_check, XA_WINDOW,
-            32, PropModeReplace, (unsigned char*)&g->stub_win, 1);
-    XChangeProperty(g->display, g->root_win, net_supporting_wm_check, XA_WINDOW,
-            32, PropModeReplace, (unsigned char*)&g->stub_win, 1);
-    XChangeProperty(g->display, g->root_win, net_supported, XA_ATOM,
-            32, PropModeReplace, (unsigned char*)supported, sizeof(supported)/sizeof(supported[0]));
+    const char *names[SUPPORTED_ATOMS + QUBES_ARRAY_SIZE(atoms_to_intern)] = {
+        "_NET_SUPPORTED",
+        "_NET_SUPPORTING_WM_CHECK",
+        /* _NET_WM_MOVERESIZE required to disable broken GTK+ move/resize fallback */
+        "_NET_WM_MOVERESIZE",
+        "_NET_WM_STATE",
+        "_NET_WM_STATE_FULLSCREEN",
+        "_NET_WM_STATE_DEMANDS_ATTENTION",
+    };
+    for (size_t i = 0; i < QUBES_ARRAY_SIZE(atoms_to_intern); ++i)
+        names[SUPPORTED_ATOMS + i] = atoms_to_intern[i].name;
+    if (!XInternAtoms(g->display, (char **)names,
+                      QUBES_ARRAY_SIZE(atoms_to_intern), False, supported)) {
+        fputs("Could not intern global atoms\n", stderr);
+        exit(1);
+    }
+    for (size_t i = 0; i < QUBES_ARRAY_SIZE(atoms_to_intern); ++i)
+        *atoms_to_intern[i].dest = supported[SUPPORTED_ATOMS + i];
+    const Atom net_supported = supported[0], net_supporting_wm_check = supported[1];
+    if (XChangeProperty(g->display, g->stub_win, g->net_wm_name, g->utf8_string_atom,
+            8, PropModeReplace, (unsigned char*)"Qubes", 5) != Success ||
+        XChangeProperty(g->display, g->stub_win, net_supporting_wm_check, XA_WINDOW,
+            32, PropModeReplace, (unsigned char*)&g->stub_win, 1) != Success ||
+        XChangeProperty(g->display, g->root_win, net_supporting_wm_check, XA_WINDOW,
+            32, PropModeReplace, (unsigned char*)&g->stub_win, 1) != Success ||
+        XChangeProperty(g->display, g->root_win, net_supported, XA_ATOM,
+            32, PropModeReplace, (unsigned char*)supported, SUPPORTED_ATOMS) != Success) {
+        fputs("Cannot change X properties\n", stderr);
```

Comment:
Result of manual test: this fails.
I don't see return value of `XChangeProperty` documented, but looking at Xlib sources, that function has a single return statement: `return 1`. It cannot return anything else, so this check is pointless.